### PR TITLE
fix various sum related warnings/errors for noncompounded returns, closes #19

### LINKED ISSUE #19: Errors and warnings with noncompounded returns and sum() type functions
I recently ran into similar warnings/errors for noncompounded returns similar to #11 (see below). The fixes are fairly straightforward, I will submit a pull request.

```
/path/to/python/lib/python3.12/site-packages/numpy/core/fromnumeric.py:86: FutureWarning: The behavior of DataFrame.sum with axis=None is deprecated, in a future version this will reduce over both axes and return a scalar. To retain the old behavior, pass axis=0 (or do not pass axis)
```
```
  File "/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/wrappers.py", line 555, in yearly_returns
    returns = returns.resample("YE").apply(_df.sum)
[...]
  File "/path/to/python/lib/python3.12/site-packages/pandas/core/frame.py", line 11657, in sum
    result = super().sum(axis, skipna, numeric_only, min_count, **kwargs)
             ^^^^^^^^^^^
TypeError: super(type, obj): obj must be an instance or subtype of type
```
```
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:426: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  .apply(apply_fnc)
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:432: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  returns.fillna(0).resample(resample).apply(apply_fnc).resample(resample).last()
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:1008: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  port["Weekly"] = port["Daily"].resample("W-MON").apply(apply_fnc)
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:1011: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  port["Monthly"] = port["Daily"].resample("ME").apply(apply_fnc)
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:1014: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  port["Quarterly"] = port["Daily"].resample("QE").apply(apply_fnc)
/path/to/python/lib/python3.12/site-packages/quantstats_lumi/_plotting/core.py:1017: FutureWarning: The provided callable <function sum at 0x7fb51bb2f2e0> is currently using DatetimeIndexResampler.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  port["Yearly"] = port["Daily"].resample("YE").apply(apply_fnc)
```

### PR DESCRIPTION
**Pull request by kartiksubbarao**

### CODE REVIEW

File: quantstats_lumi/stats.py [hunk 1]
```diff
@@ -79,7 +79,7 @@ def get_outliers(data):
         else:
             returns = returns[returns.columns[0]]
 
-    apply_fnc = comp if compounded else _np.sum
+    apply_fnc = comp if compounded else 'sum'
```
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/stats.py:82]: @korbit-ai-mentor You are incorrect. apply_fnc is passed as an argument to apply() which can take string or function arguments.

File: quantstats_lumi/stats.py [hunk 2]
```diff
@@ -594,7 +594,7 @@ def cagr(returns, rf=0.0, compounded=True, periods=365):
     if compounded:
         total = comp(total)
     else:
-        total = _np.sum(total)
+        total = _np.sum(total, axis=0)
```
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/stats.py:597]: @korbit-ai-mentor Incorrect. The code works as written with both DataFrames and Series. There are various ways to optimize the code, but I wanted to keep the change simple and not spend a lot of time on the fix.

File: quantstats_lumi/_plotting/wrappers.py [hunk 3]
```diff
@@ -552,7 +551,7 @@ def yearly_returns(
     if compounded:
         returns = returns.resample("YE").apply(_stats.comp)
     else:
-        returns = returns.resample("YE").apply(_df.sum)
+        returns = returns.resample("YE").sum()
```
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor The code works as written.

[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor No thanks bot. I don't think it's necessary.

File: quantstats_lumi/stats.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/stats.py:82]: @korbit-ai-mentor Read the code, apply_fnc is only used a few times right below where it is set. There is no confusion here at all. The pandas warning expressly recommends using a string and **not** a callable.

File: quantstats_lumi/stats.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/stats.py:597]: @korbit-ai-mentor No thanks bot. I don't think it's necessary.

File: quantstats_lumi/_plotting/wrappers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor Currently there is no testing framework for this codebase. The maintainer can choose to implement one if he desires. The purpose of this patch is to fix a very specific and minor issue, not to introduce a new testing framework.

File: quantstats_lumi/stats.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/stats.py:597]:
```
(Pdb) df = pd.DataFrame({ 'A': np.random.rand(3), 'B': np.random.rand(3) })
(Pdb) df
          A         B
0  0.599800  0.437393
1  0.059790  0.948146
2  0.990512  0.256326
(Pdb) np.sum(df, axis=0)
A    1.650102
B    1.641865
dtype: float64
(Pdb) type(df.A)
<class 'pandas.core.series.Series'>
(Pdb) np.sum(df.A, axis=0)
1.6501016169769844
```

File: quantstats_lumi/_plotting/wrappers.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor No, that's not necessary for this patch.

[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor No.

[the remark this answers is not in the document]
Comment (kartiksubbarao) [reply quantstats_lumi/_plotting/wrappers.py:554]: @korbit-ai-mentor No. No more action is needed on this patch, do not bring up any more suggestions.

**Review by grzesir — APPROVED** [review]
Thank you for your contributions. We will include this in the next release

**Comment by kartiksubbarao** [comment]
Thanks @grzesir!

**Comment by grzesir** [comment]
No problem. I can deploy sooner if you think it's something important. Otherwise planning another release in 2-3 weeks

**Comment by kartiksubbarao** [comment]
@grzesir No rush, I've patched locally and can wait for the next release to update from pypi.

**Comment by grzesir** [comment]
Ok sounds good. Will release it with our other updated in a few weeks.
We're planning an upgrade that will include the strategy parameters in the
tearsheet from lumibot backtests

Robert Grzesik 347-635-3416

**Pull request merged**